saleor/plugins/avatax/plugin.py

```diff
@@ -636,10 +636,17 @@ def calculate_order_shipping(
                     gross = Money(amount=net + tax, currency=currency)
                     net = Money(amount=net, currency=currency)
                 return TaxedMoney(net=net, gross=gross)
+
+        # Ignore typing checks because it is checked in _validate_order
```

Comment:
Do we really need this part? Maybe we could simply extend `saleor.plugins.avatax._validate_order` to return False when `channel_listing ` is not available? - As we can say: 'we don't have proper shipping method assigned. This order doesn't have enough info to fetch taxes for it`
Also, do we need to raise an exception? I thought that in case of removing channel_listing for a given method, Saleor will block a possibility to finalize an order, no need to double-check it inside a Tax method, right?

Reply:
But in this situation, the `previous_value` will be returned and an exception will be not raised, I am not sure if it's the correct solution.